0010-zkapp-cli-lightnet-management.md

```diff
@@ -0,0 +1,100 @@
+  - --archive-node: `true` or `false` (default: `true`)
+  - --mina-branch: `rampup`, `berkeley` or `develop` (default: `rampup`)
+  - --dune-profile: `lightnet` or `devnet` (default: `lightnet`)
+- **Output**:
+  - Anticipated network properties (as per [Docker image description](https://hub.docker.com/r/o1labs/mina-local-network));
+  - Mina Daemon GraphQL endpoint for communication: [http://localhost:8080/graphql](http://localhost:8080/graphql)
+  - PostgreSQL connection string: `postgresql://postgres:postgres@localhost:5432/archive`;
+  - Mina Daemon and Archive Node logs path;
+  - [o1js configuration](https://github.com/o1-labs/o1js/blob/ccf50e0b58190d9700c6dda4fafee3e62e270131/src/examples/zkapps/hello_world/run_live.ts#L18) code snippet.
+
+#### Stop lightnet network
+
+- **Command**: `zk lightnet stop`
+  - In future we might want to extend this command with additional options to preserve the network state between containers lifecycle.
+- **Function**: Destroys the running container. Repeating the command while network is down should not throw an error but rather output a message that the network is already down.
+- **Output**: Maybe blockchain state (blocks height, slots, epoch, txns processed during the session, etc.) at the time of command issuing.
+
+#### Handle logs
+
+- **Command**: `zk lightnet logs <sub-command>`
+- **Sub-commands**:
+  - **view**: `less` or `tail` the log files (Mina Daemon or Archive Node). OS agnostic.
+  - **save**: Saves the log files to default location and outputs the resulting path.
+
+#### Manage accounts
+
+- **Command**: `zk lightnet account <sub-command>`
+- **Sub-commands**:
+  - **fetch**: Fetches available account from the container's pool.
+  - **release**: Releases "used" account back to the pool.
+- **Note**: Corresponding [o1js API](https://github.com/o1-labs/o1js/pull/1167) can be used.
+
+#### Launch the blockchain explorer
+
+- **Command**: `zk lightnet explorer`
+- **Function**: Opens the local lightweight blockchain explorer (HTML + JS + CSS application) in default browser tab.
+  - Existing lightweight explorer will be refactored and ported under the [zkapp-cli](https://github.com/o1-labs/zkapp-cli) repository.
+- **Output**: The URL of application.
+
+### Impact
+
+This update enhances the existing zkapp CLI by appending more commands under the `lightnet` sub-command. The core operations will remain intact.
+
+## Test plan and functional requirements
+
+- **Testing goals and objectives**:
+  - To validate the reliability of the enhanced zkapp CLI tool.
+- **Testing approach**:
+  - Unit tests for the new `lightnet` subcommands.
+  - Integration tests for the new `lightnet` subcommands.
+- **Testing scope**:
+  - Testing all lightnet commands and their possible variations.
+  - Addressing potential error scenarios and recovery mechanisms.
+- **Testing requirements**:
+  - Test in various local setups to simulate different development environments.
```

Comment:
In the future it may be useful to test interacting with the light network from a browser UI.

Reply:
Agreed that this is a future requirement but I would love to see this happen at some point.

Reply:
To be fair I need to say that the `light network` is already in use for `zkapp-cli` and `o1js` E2E tests. This RFC and implementation is essentially the wrapper around https://hub.docker.com/r/o1labs/mina-local-network and other tools that is already successfully in use and works really good (including local env. and CI).
Next [o1js](https://github.com/o1-labs/o1js/issues/1169) and [zkapp-cli](https://github.com/o1-labs/zkapp-cli/issues/478) tests will continue utilize same toolset as it is right now (check CI jobs of repos mentioned).
Should we add more capabilities to `zkapp-cli` and allow users interact with deployed zkApp via UI? No problem, since we already have all the tools.